src/Features/Core/Portable/Snippets/RoslynLSPSnippetConverter.cs

```diff
@@ -0,0 +1,174 @@
+                }
+            }
+
+            return map;
+        }
+
+        /// <summary>
+        /// Tries to see if a value exists at that position in the map, and if so it
+        /// generates a string that is LSP formatted as well as passes back the length
+        /// of the identifier so that it can skip forward in the string.
+        /// </summary>
+        private static (string str, int strLength) GetStringInPosition(Dictionary<int, (string identifier, int priority)> map, int position)
+        {
+            if (map.TryGetValue(position, out var placeholderInfo))
+            {
+                return ($"${{{placeholderInfo.priority}:{placeholderInfo.identifier}}}", placeholderInfo.identifier.Length);
+            }
+
+            return (string.Empty, 0);
+        }
+
+        /// <summary>
+        /// We need to extend the snippet's TextChange if any of the placeholders or
+        /// if the caret position comes before or after the span of the TextChange.
+        /// If so, then find the new string that encompasses all of the placeholders
+        /// and caret position.
+        /// </summary>
+        private static async Task<TextChange> ExtendSnippetTextChangeAsync(Document document, TextChange textChange, ImmutableArray<SnippetPlaceholder> placeholders, int caretPosition)
+        {
+            var extendedSpan = GetUpdatedTextSpan(textChange, placeholders, caretPosition);
+
+            if (extendedSpan.Length == 0)
+            {
+                return textChange;
+            }
+
+            var documentText = await document.GetTextAsync().ConfigureAwait(false);
+            var newString = documentText.ToString(extendedSpan);
+            var newTextChange = new TextChange(new TextSpan(extendedSpan.Start, 0), newString);
+
+            return newTextChange;
+        }
+
+        /// <summary>
+        /// Iterates through the placeholders and determines if any of the positions
+        /// come before or after what is indicated by the snippet's TextChange.
+        /// If so, adjust the starting and ending position accordingly.
+        /// </summary>
+        private static TextSpan GetUpdatedTextSpan(TextChange textChange, ImmutableArray<SnippetPlaceholder> placeholders, int caretPosition)
+        {
+            var textSpanLength = textChange.NewText!.Length;
+
+            var startPosition = textChange.Span.Start;
+            var endPosition = textChange.Span.Start + textSpanLength;
+
+            foreach (var placeholder in placeholders)
+            {
+                foreach (var position in placeholder.PlaceHolderPositions)
+                {
+                    if (startPosition > position)
+                    {
+                        startPosition = position;
+                    }
+
+                    if (endPosition < position)
+                    {
+                        endPosition = position;
+                    }
```

Comment:
```suggestion
                    startPosition = Math.Min(startPosition , position);
                    endPosition = Math.Max(endPosition, position);
```

(do the same for the caret position check).

Reply:
also, you can use MaxBy/MinBy here for effectively: `startPosition = Math.Min(startPosition, placeholders.MinBy(h => h.PlaceHolderPositions.Min());` etc.